lldb/source/Target/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -991,20 +995,221 @@ llvm::Optional<uint64_t> SwiftLanguageRuntimeImpl::GetMemberVariableOffset(
+    std::string override_name;
+    if (rti->getRecordKind() == swift::reflection::RecordKind::Tuple) {
+      override_name = ts->GetTupleElementName(type.GetOpaqueQualType(), idx);
+      if (override_name.empty())
+        LLDB_LOGF(GetLogIfAllCategoriesSet(LIBLLDB_LOG_TYPES),
+                  "couldn't determine name for tuple element %zu in %s", idx,
+                  type.GetTypeName().AsCString());
+    }
+    return get_from_field_info(fields[idx], override_name);
+  }
+  // Enums.
+  if (auto *eti = llvm::dyn_cast_or_null<swift::reflection::EnumTypeInfo>(ti)) {
+    unsigned i = 0;
+    for (auto &enum_case : eti->getCases()) {
+      // Skip non-payload cases.
+      if (!enum_case.TR)
+        continue;
+      if (i++ == idx)
+        return get_from_field_info(enum_case, "");
+    }
+    LLDB_LOGF(GetLogIfAllCategoriesSet(LIBLLDB_LOG_TYPES),
+              "index %zu is out of bounds (%d)", idx,
+              eti->getNumPayloadCases());
+    return {};
+    /*
+    if (element_info->is_indirect) {
+      language_flags |= LanguageFlags::eIsIndirectEnumCase;
+      return ToCompilerType(GetASTContext()->TheRawPointerType.getPointer());
+      } else
+        return element_info->payload_type;
+    */
+  }
+
+  // Objects.
+  // Try the instance type metadata.
+  if (!valobj)
+    return {};
+  bool found_start = false;
+  swift::Demangle::Demangler dem;
+  auto mangled = type.GetMangledTypeName().GetStringRef();
+  NodePointer type_node = dem.demangleSymbol(mangled);
+  llvm::StringRef type_name = GetBaseName(type_node);
+  llvm::SmallVector<SuperClassType, 2> supers;
+  ForEachSuperClassType(*valobj, [&](SuperClassType sc) -> bool {
+    if (!found_start) {
+      // The ValueObject always points to the same class instance,
+      // even when querying base classes. Drop base classes until we
+      // reach the requested type.
+      auto *type_ref = sc.get_typeref();
+      swift::Demangle::NodePointer base_class = type_ref->getDemangling(dem);
+      if (GetBaseName(base_class) != type_name)
+        return false;
+      found_start = true;
+    }
+    supers.push_back(sc);
+    return supers.size() >= 2;
+  });
+
+  if (supers.size() == 0) {
+    LLDB_LOGF(GetLogIfAllCategoriesSet(LIBLLDB_LOG_TYPES),
+              "Couldn't find the type metadata for %s in instance",
+              type.GetTypeName().AsCString());
+    return {};
+  }
+
+  auto *object = supers[0].get_record_type_info();
+  if (!object)
+    return {};
+
+  LLDB_LOGF(GetLogIfAllCategoriesSet(LIBLLDB_LOG_TYPES),
+            "using instance type info");
+
+  // Handle the artificial base class fields.
+  unsigned i = 0;
+  if (supers.size() > 1) {
+    auto *type_ref = supers[1].get_typeref();
+    auto *objc_tr =
+        llvm::dyn_cast_or_null<swift::reflection::ObjCClassTypeRef>(type_ref);
+    // SwiftASTContext hides the ObjC base class for Swift classes.
+    if (!objc_tr || objc_tr->getName() != "_TtCs12_SwiftObject")
```

Comment:
Is this old-mangled baked into the ABI? I'm surprised to see an old mangled symbol in new code.